test/buttercup-tests/flycheck/flycheck-test.el

```diff
@@ -0,0 +1,23 @@
+(describe "Flycheck integration"
+  (it "integrates flycheck with omnisharp"
+    (ot--open-the-minimal-solution-source-file "FlycheckTest.cs")
+    (ot--buffer-contents-and-point-at-$
+     "$public class MyClass {
+         DoesNotExist foo;
+       }")
+
+    (flycheck-mode)
+    (flycheck-list-errors)
+    ;; (ot--wait-for (lambda() (print "waiting")(get-buffer "*Flycheck errors*")) 5)
+    (ot--wait-for (lambda() (and (boundp 'flycheck-current-errors)
+                                 (> (length flycheck-current-errors) 0))) 10)
```

Comment:
Sure, I can take a look at it. Maybe flycheck itself has some trick for
running tests reliably.
4.8.2015 5.08 ap. "Jason Imison" <notifications@github.com> kirjoitti:

> In test/buttercup-tests/flycheck/flycheck-test.el
> <https://github.com/OmniSharp/omnisharp-emacs/pull/191#discussion_r36151936>
> :
>
> > @@ -0,0 +1,23 @@
> > +(describe "Flycheck integration"
> > +  (it "integrates flycheck with omnisharp"
> > +    (ot--open-the-minimal-solution-source-file "FlycheckTest.cs")
> > +    (ot--buffer-contents-and-point-at-$
> > +     "$public class MyClass {
> > +         DoesNotExist foo;
> > +       }")
> > +
> > +    (flycheck-mode)
> > +    (flycheck-list-errors)
> > +    ;; (ot--wait-for (lambda() (print "waiting")(get-buffer "*Flycheck errors*")) 5)
> > +    (ot--wait-for (lambda() (and (boundp 'flycheck-current-errors)
> > +                                 (> (length flycheck-current-errors) 0))) 10)
>
> This line never finds any errors when ran from the integration tests
> script - I can't figure out why. I tried quite a few other ways of
> achieving the same thing but no luck.
> It's odd - the commented out look for 'Flycheck buffer' line above doesn't
> fail, so I can see that flycheck mode is running.
>
> The test passes every time when ran locally using M-x
> buttercup-run-at-point
>
> @sp3ctum <https://github.com/sp3ctum> Any chance that you could take a
> look at this for me? I'd appreciate it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OmniSharp/omnisharp-emacs/pull/191/files#r36151936>.
>